liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/server/DevMojo.java

```diff
@@ -0,0 +1,889 @@
+
+        Plugin plugin = project.getPlugin(groupId + ":" + artifactId);
+        if (plugin == null) {
+            plugin = plugin(groupId(groupId), artifactId(artifactId), version("RELEASE"));
+        }
+        Xpp3Dom config = null;
+
+        List<Plugin> buildPlugins = project.getBuildPlugins();
+        for (Plugin p : buildPlugins) {
+            if (p.equals(plugin)) {
+                config = (Xpp3Dom) p.getConfiguration();
+                
+                PluginExecution pe;
+                Map<String, PluginExecution> peMap = p.getExecutionsAsMap();
+                if ((pe = peMap.get("default-" + phase)) != null || (pe = peMap.get(phase)) != null
+                        || (pe = peMap.get("default")) != null) {
+                    Xpp3Dom executionConfig = (Xpp3Dom) pe.getConfiguration();
+                    config = Xpp3Dom.mergeXpp3Dom(executionConfig, config);
+                }
+                break;
+            }
+        }
+
+        if (config == null) {
+            log.debug("Could not find " + artifactId + " configuration for " + phase
+                    + " phase. Creating new configuration.");
+            config = configuration();
+        }
+
+        if (phase.equals("test")) {
+            injectTestId(config);
+        } else if (phase.equals("integration-test")) {
+            injectTestId(config);
+            // clean up previous summary file
+            File summaryFile = null;
+            Xpp3Dom summaryFileElement = config.getChild("summaryFile");
+            if (summaryFileElement != null) {
+                summaryFile = new File(summaryFileElement.getValue());
+            } else {
+                summaryFile = new File(project.getBuild().getDirectory() + "/failsafe-reports/failsafe-summary.xml");
+            }
+            try {
+                log.debug("Looking for summary file at " + summaryFile.getCanonicalPath());
+            } catch (IOException e) {
+                log.debug("Unable to resolve summary file " + e.getMessage());
+            }
+            if (summaryFile.exists()) {
+                boolean deleteResult = summaryFile.delete();
+                log.debug("Summary file deleted? " + deleteResult);
+            } else {
+                log.debug("Summary file doesn't exist");
+            }
+        }
+        log.debug(artifactId + " configuration for " + phase + " phase: " + config);
+
+        executeMojo(plugin, goal(phase), config, executionEnvironment(project, session.clone(), pluginManager));
+    }
+
+    /**
+     * Force change a property so that the checksum calculated by
+     * AbstractSurefireMojo is different every time.
+     *
+     * @param config
+     *            The configuration element
+     */
+    private void injectTestId(Xpp3Dom config) {
+        Xpp3Dom properties = config.getChild("properties");
+        String propertyName = "liberty.dev.test.run.id";
```

Comment:
This could be a constant (static final at class level).

Reply:
Extracted to class level